File: solutions/0003-longest-substring-without-repeating-characters/solution.java
class Solution {
    public int lengthOfLongestSubstring(String s) {

        int length = s.length();

        if (length == 0) return 0;

        if (length == 1) return 1;

        int max = -1;

        for(int i = 0; i < length; i++) {

            boolean flag = true;

            if (max >= length - i)
                return max;

            for(int j = i+1; j < length; j++) {

                String str = s.substring(i, j);

                if (str.indexOf(s.charAt(j)) != -1) {

                    if(str.length() > max) {
                        max = str.length();
                    }

                    flag = false;

                    break;
                }
            }

            if (flag && length - i > max) {
                max = length - i;
            }
        }

        return max;
    }
}

File: solutions/0003-longest-substring-without-repeating-characters/solution.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:

        n = len(s)
        if n == 0: 
            return 0
        if n == 1:
            return 1

        max = -1    

        for i in range (n):
            flag = True

            if(max >= n - i):
                return max

            for j in range (i+1, n):
                substr = s[i:j]
                
                if(substr.find(s[j]) != -1):
                    if(len(substr) > max):
                        max = len(substr)
                    flag = False    
                    break    
                
            if flag and n - i > max:
                return n - i
        
        return max
